guide.md

```diff
@@ -0,0 +1,388 @@
+Getting Started 2.0
+===
+
+For this guide, we're going to walk through some simple examples for each of the
+pieces of CRUD. Each step in this guide will build on the previous, and is meant
+to be followed along. This guide assumes that you're using PostgreSQL on
+nightly. We'll talk about how to use Diesel on stable in the final chapter.
+Before we start, make sure you have PostgreSQL installed and running.
+
+The first thing we need to do is generate our project.
+
+```shell
+cargo new diesel_demo
+cd diesel_demo
+```
+
+We're going to use a [`.env`][dotenv-rust] to manage our environment variables
+for us, and [`diesel_cli`][diesel-cli] to help set up the database.
+
+[dotenv-rust]: https://github.com/slapresta/rust-dotenv
+[diesel-cli]: https://github.com/sgrif/diesel/diesel_cli
+
+```shell
+echo DATABASE_URL=postgres://localhost/diesel_demo > .env
+cargo install diesel_cli
+diesel setup
+```
+
+This will create our database for us (if it didn't already exist), and create an
+empty migrations directory that we can use to manage our schema. More on that
+later. Let's add Diesel to our `Cargo.toml`. We'll also add `dotenv` so we can
+use that same `.env` file for our environment variables.
+
+```toml
+[dependencies]
+diesel = "^0.5.0"
+diesel_codegen = { version = "^0.5.0", default-features = false, features = ["nightly", "postgres"] }
+dotenv = "^0.8.0"
+dotenv_macros = "^0.8.0"
+```
+
+We'll also add this to the top of `src/lib.rs`, which will allow us to use
+features from `diesel_codegen`.
```

Comment:
I'm open to suggestions on this one. codegen being a separate crate might be fairly alien to devs from other languages. I don't think I ever really cover what it does, and I don't have a great "quick summary" other than "it generates code".